src/Response/Segment/CreateSegmentResponse.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OneSignal\Response\Segment;
+
+use OneSignal\Response\AbstractResponse;
+
+class CreateSegmentResponse implements AbstractResponse
+{
+    protected bool $success;
+
+    /**
+     * @var non-empty-string
+     */
+    protected string $id;
+
+    /**
+     * @param non-empty-string $id
+     */
+    public function __construct(bool $success, string $id)
```

Comment:
What's the point of `success` ? I'm thinking about to differentiate response objects per status code, because for me this looks like "rest api where every response is 200 with different data" :D

Reply:
That's the problem  - that I don't like current implemented solution where status code is injected into response array for people to be able to detect errors :(

Reply:
you really don't need to be worried by that... for 99% cases that will be always `true`
and if something breaks you always have Guzzle request exceptions in the request stage itself
to me, this is completly ok

Reply:
But then if we let users to rely on exceptions they have to parse the request bodies themself :) We could give nice response objects with parsed list of errors, WDYT?

Reply:
By the way from `psr/http-client`:
> A Client MUST NOT treat a well-formed HTTP request or HTTP response as an error condition. For example, response status codes in the 400 and 500 range MUST NOT cause an exception and MUST be returned to the Calling Library as normal.

So that means it would try to create our successful response model from responses with 400-500 status code

Reply:
Ok so, if we get non successful status code, then we throw our exception with included request, response, otherwise always return response model, easiest compromise.

Reply:
check it out